gloom/gloom/src/infrastructure/graphics/engine/GlGraphicsEngine.h

```diff
@@ -25,27 +22,26 @@ class GlGraphicsEngine : public GraphicsEngine
 	void StartFrame() override;
 	void EndFrame() override;
 
-	std::unique_ptr<Shader> CreateShader() override;
-	std::unique_ptr<Shader> ImportShader(const std::string& filePath) override;
-
 	void RegisterLight(DirectionalLight& directionalLight) override;
 	void RegisterLight(PointLight& pointLight) override;
 
 	void Render(const Camera&, Mesh& mesh) override;
 	void Render(const Camera&, Skybox& skybox) override;
 	void Render(const Camera&, Sprite& sprite) override;
+
+	const MaterialTemplate* GetMaterialTemplate(const Shader& shader) override;
```

Comment:
Investigate if this could return `const MaterialTemplate&` instead.

Reply:
Decided to leave it like that. I don't know how the other rendering APIs like DirectX will manage the material templates, so I prefer the loose approach here.